### PR TITLE
Convert complex CSS files to SCSS

### LINKED ISSUE #374: ✨ [Feature request]: Convert complex CSS files to SCSS
There are a bunch of CSS files which are complex in nature and were written in the initial days. Convert them to SCSS for better managing.

### PR DESCRIPTION
**Pull request by 6km**
# Description

Convert complex CSS files to SCSS

Fixes #374 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Locally

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 6km** [comment]
@nirmalkc can you review?

**Comment by atapas** [comment]
@6km Please resolve merge conflicts.

**Comment by 6km** [comment]
> @6km Please resolve merge conflicts.

done

**Comment by nirmalkc** [comment]
@6km and @atapas 

Here is my recommendation about the SASS file structure. This approach will be much easier to maintain and reuse in the long run.

src/
|
|- base /
|    |- _variables.scss   // All CSS 4 Variables
|    |- _typography.scss  // Typography rules
|
|- components / 
|    |- _buttons.scss     // Buttons
|    |- _indicators.scss  // Spinner, progress bar, preloaders etc.,
|
|– layout/
|   |– _navigation.scss   // Navigation
|   |– _header.scss       // Header
|   |– _search.scss       // Search
|
|– pages/
|   |– _home.scss         // Home page styles
|   |– _playideas.scss    // Playideas
|   |– _playlist.scss     // Contact specific styles
|
|- App.scss

Though we are in SCSS eco-system, lets continue to use CSS4 Variables for any direct exposure of visual properties.

**Comment by nirmalkc** [comment]
@atapas , should we continue to rely on Google Font repo for direct font include or shall we keep the fonts in our local git folder, so that we dont rely on them anymore? I am fine with either way, but keeping it in our local repo-reduces the dependency.

`@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600&family=Raleway:wght@900&display=swap');`

**Comment by atapas** [comment]
> @atapas , should we continue to rely on Google Font repo for direct font include or shall we keep the fonts in our local git folder, so that we dont rely on them anymore? I am fine with either way, but keeping it in our local repo-reduces the dependency.
> 
> `@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600&family=Raleway:wght@900&display=swap');`

We can keep them in local.

File: src/common/header/header.scss [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
... fill='none' stroke='%2300f2fe' stroke-linecap='round' stroke-width='2.6'%3E%3Cellipse cx='20.579' cy='8.689' rx='20.579' ry='8.689' stroke='none'/%3E%3Cellipse cx='20.579' cy='8.689' rx='19.279' ry='7.389' fill='none'/%3E%3C/g%3E%3C/g%3E%3C/g%3E%3C/g%3E%3Cpath id='Path_1' data-name='Path 1' d='M408.4 30.28l-4.563 7.985 4.563 6.274h7.7l4.706-6.844L416.81 30.28Z' transform='translate(-11.246 -2.736)' fill='%2361dafb'/%3E%3Cpath id='Polygon_2' data-name='Polygon 2' d='M26.2 12.227a2.645 2.645 0 0 1 4.523 2.6l-.018.048a2.645 2.645 0 0 0 2.04 3.529l.087.015a2.645 2.645 0 0 1 0 5.216l-.087.015a2.645 2.645 0 0 0-2.04 3.529l.018.048a2.645 2.645 0 0 1-4.523 2.6h0a2.645 2.645 0 0 0-4.089 0h0a2.645 2.645 0 0 1-4.523-2.6l.018-.048a2.645 2.645 0 0 0-2.04-3.529l-.087-.015a2.645 2.645 0 0 1 0-5.216l.087-.015a2.645 2.645 0 0 0 2.04-3.529l-.018-.048a2.645 2.645 0 0 1 4.523-2.6h0a2.645 2.645 0 0 0 4.089 0Z' transform='translate(390.665 4.291) rotate(30)' fill='%2300f2fe'/%3E%3Cpath id='Polygon_3' data-name='Polygon 3' d='M5.689 1.122a1 1 0 0 1 1.621 0l4.544 6.292A1 1 0 0 1 11.044 9H1.956a1 1 0 0 1-.811-1.585Z' transform='translate(407 27.544) rotate(90)' fill='%23010326'/%3E%3C/g%3E%3C/svg%3E"); */
```
Comment (nirmalkc) [inline src/common/header/header.scss:34]: @6km , the commented code can be removed.

Comment (6km) [reply src/common/header/header.scss]: done

**Comment by nirmalkc** [comment]
> @6km and @atapas
> 
> Here is my recommendation about the SASS file structure. This approach will be much easier to maintain and reuse in the long run.
> 
> src/ | |- base / | |- _variables.scss // All CSS 4 Variables | |- _typography.scss // Typography rules | |- components / | |- _buttons.scss // Buttons | |- _indicators.scss // Spinner, progress bar, preloaders etc., | |– layout/ | |– _navigation.scss // Navigation | |– _header.scss // Header | |– _search.scss // Search | |– pages/ | |– _home.scss // Home page styles | |– _playideas.scss // Playideas | |– _playlist.scss // Contact specific styles | |- App.scss
> 
> Though we are in SCSS eco-system, lets continue to use CSS4 Variables for any direct exposure of visual properties.

@atapas and @6km have we decided to keep the SCSS files within the respective folders like header, platlist etc.,? If yes, then we are good to merge.

**Comment by atapas** [comment]
> > @6km and @atapas
> > Here is my recommendation about the SASS file structure. This approach will be much easier to maintain and reuse in the long run.
> > src/ | |- base / | |- _variables.scss // All CSS 4 Variables | |- _typography.scss // Typography rules | |- components / | |- _buttons.scss // Buttons | |- _indicators.scss // Spinner, progress bar, preloaders etc., | |– layout/ | |– _navigation.scss // Navigation | |– _header.scss // Header | |– _search.scss // Search | |– pages/ | |– _home.scss // Home page styles | |– _playideas.scss // Playideas | |– _playlist.scss // Contact specific styles | |- App.scss
> > Though we are in SCSS eco-system, lets continue to use CSS4 Variables for any direct exposure of visual properties.
> 
> @atapas and @6km have we decided to keep the SCSS files within the respective folders like header, platlist etc.,? If yes, then we are good to merge.

@nirmalkc I would look for suggestions here. What would be best for the long run? Please remember, the plays will go to a separate repository soon. So its better if the respective artefacts stay close by.

**Comment by nirmalkc** [comment]
> > > @6km and @atapas
> > > Here is my recommendation about the SASS file structure. This approach will be much easier to maintain and reuse in the long run.
> > > src/ | |- base / | |- _variables.scss // All CSS 4 Variables | |- _typography.scss // Typography rules | |- components / | |- _buttons.scss // Buttons | |- _indicators.scss // Spinner, progress bar, preloaders etc., | |– layout/ | |– _navigation.scss // Navigation | |– _header.scss // Header | |– _search.scss // Search | |– pages/ | |– _home.scss // Home page styles | |– _playideas.scss // Playideas | |– _playlist.scss // Contact specific styles | |- App.scss
> > > Though we are in SCSS eco-system, lets continue to use CSS4 Variables for any direct exposure of visual properties.
> > 
> > 
> > @atapas and @6km have we decided to keep the SCSS files within the respective folders like header, platlist etc.,? If yes, then we are good to merge.
> 
> @nirmalkc I would look for suggestions here. What would be best for the long run? Please remember, the plays will go to a separate repository soon. So its better if the respective artefacts stay close by.

Play specific scss can be bundled in the same folder. Whereas the platform code can be structured as per my above recommendation, atleast the base, components and layout part.

**Comment by nirmalkc** [comment]
@6km , if can you please take care of the file structure changes, I think we are good to merge.

**Comment by nirmalkc** [comment]
> > @6km and @atapas
> > Here is my recommendation about the SASS file structure. This approach will be much easier to maintain and reuse in the long run.
> > src/ | |- base / | |- _variables.scss // All CSS 4 Variables | |- _typography.scss // Typography rules | |- components / | |- _buttons.scss // Buttons | |- _indicators.scss // Spinner, progress bar, preloaders etc., | |– layout/ | |– _navigation.scss // Navigation | |– _header.scss // Header | |– _search.scss // Search | |– pages/ | |– _home.scss // Home page styles | |– _playideas.scss // Playideas | |– _playlist.scss // Contact specific styles | |- App.scss
> > Though we are in SCSS eco-system, lets continue to use CSS4 Variables for any direct exposure of visual properties.
> 
> I agree with you, I think that this will be better to maintain and easier to understand for new contributors.
> 
> I will work on it ASAP.

@6km , can we atleast bring the common areas like the variables, typography, buttons in separate SCSS?

**Comment by 6km** [comment]
> @6km , can we atleast bring the common areas like the variables, typography, buttons in separate SCSS?

I'm working on the new file structure right now, sorry for being late

**Comment by 6km** [comment]
@nirmalkc I have been working on the new file structure, can you re-review please?

**Comment by 6km** [comment]
Btw, I have replaced node-sass with sass, as it's currently deprecated.

image

**Comment by koustov** [comment]
@nirmalkc can you taka a look?

**Comment by atapas** [comment]
@6km resolve merge conflicts

**Comment by 6km** [comment]
> @6km resolve merge conflicts

done

**Comment by 6km** [comment]
can anyone review please?

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
> can anyone review please?

@6km It's ready. I just want someone to test it well and confirm.

**Comment by 6km** [comment]
> @6km It's ready. I just want someone to test it well and confirm.

@atapas Great! 👍

**Comment by atapas** [comment]
Hi @6km can you please resolve the conflicts?

@vasantisuthar as you are testing it, please provide your updates. We want to merge it asap as the changes will lead to merge conflicts very often.

**Comment by vasantisuthar** [comment]
> Hi @6km can you please resolve the conflicts?
> 
> @vasantisuthar as you are testing it, please provide your updates. We want to merge it asap as the changes will lead to merge conflicts very often.

Will update you tomorrow

**Comment by 6km** [comment]
> > Hi @6km can you please resolve the conflicts?
> > @vasantisuthar as you are testing it, please provide your updates. We want to merge it asap as the changes will lead to merge conflicts very often.
> 
> Will update you tomorrow

I have wrongfully merged this PR with 0 commits, I will open another one tomorrow, please be ready to review it 🙏

**Pull request merged**